### PR TITLE
ESTS-136762 Added new fields for node to get form Node-discover

### PR DESCRIPTION
**Pull request by arunrordell**
This PR will gets additional info of a discovered node from node-discovery paqx

### CODE REVIEW

**Comment by arunrordell** [comment]
Dependency PRs

https://github.com/dellemc-symphony/rackhd-packaging/pull/6

https://github.com/dellemc-symphony/rackhd-adapter-parent/pull/255

https://github.com/dellemc-symphony/node-discovery-paqx-parent/pull/109

https://github.com/dellemc-symphony/compute-capabilities-api/pull/105

**Comment by arunrordell** [comment]
dellemc-symphony/compute-capabilities-api#105 should be merged before other dependent PRs

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java [hunk 1]
```diff
@@ -443,25 +445,20 @@ public void executeRequest(String requestId) throws Exception
...
+        {
+            LOGGER.info("There is no Discovered Nodes");
         }
 
+        DiscoveredNode discoveredNode =  discoveredNodes.get(0);
+        DiscoveredNodeInfo discoveredNodeInfo = new DiscoveredNodeInfo(discoveredNode.getProduct(),null,discoveredNode.getProduct(),null,discoveredNode.getSerial(),
+             discoveredNode.getConvergedUuid());
+        discoveredNodeInfo.setNodeStatus(NodeStatus.valueOf(discoveredNode.getNodeStatus().toString()));
+        discoveredNodeInfo.setVendor(discoveredNode.getVendor());
+        retList.add(discoveredNodeInfo);
```
Comment (gavin-scott) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java]: Why has this been changed to only include one node?

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/adapter/NodesListedResponseAdapter.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 /**
  * List nodes Response Adapter
- *
+ * <p>
```
Comment (gavin-scott) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/adapter/NodesListedResponseAdapter.java]: Delete

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/model/NodeInfo.java [hunk 3]
```diff
@@ -32,6 +32,7 @@ public NodeInfo(String symphonyUuid, NodeStatus nodeStatus)
         this.serialNumber = "";
     }
 
+
```
Comment (gavin-scott) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/model/NodeInfo.java]: Delete

File: dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java [hunk 4]
```diff
@@ -198,10 +198,10 @@ protected void waitForServiceCallback(ServiceCallback serviceCallback, String re
 
     private List<com.dell.cpsd.DiscoveredNode> buildNodeList(String uuid)
     {
-        com.dell.cpsd.DiscoveredNode discoveredNode = new com.dell.cpsd.DiscoveredNode(uuid,
+        com.dell.cpsd.DiscoveredNode discoveredNode = new com.dell.cpsd.DiscoveredNode(uuid, "testserial", "testproduct","test_Dell Inc",
                 com.dell.cpsd.DiscoveredNode.AllocationStatus.DISCOVERED);
-        com.dell.cpsd.DiscoveredNode addedNode = new com.dell.cpsd.DiscoveredNode(UUID.randomUUID().toString(),
-                com.dell.cpsd.DiscoveredNode.AllocationStatus.ADDED);
+        com.dell.cpsd.DiscoveredNode addedNode = new com.dell.cpsd.DiscoveredNode(UUID.randomUUID().toString(), "testserial",
+                "testproduct","test_Dell Inc", com.dell.cpsd.DiscoveredNode.AllocationStatus.ADDED);
```
Comment (gavin-scott) [inline dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java]: Space after first comma

File: dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java [hunk 5]
```diff
@@ -251,64 +251,30 @@ public void testListDiscoveredNodeInfo() throws ServiceTimeoutException, Service
...
                 StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+
                 if (stackTraceElements[2].getMethodName().equalsIgnoreCase("listDiscoveredNodes"))
                 {
-                    return new ServiceResponse<>(serviceRequestCallback.getRequestId(), listed, "test");
-                }
-                if (stackTraceElements[2].getMethodName().equalsIgnoreCase("listNodeInventory"))
-                {
-                    return new ServiceResponse<>(serviceRequestCallback.getRequestId(), nodeInventoryResponseMessage, "test");
+                    ServiceResponse response =new  ServiceResponse<>(serviceRequestCallback.getRequestId(), listed, "test");
```
Comment (gavin-scott) [inline dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java]: Space after `=`

File: dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/delegates/FindDiscoveredNodesTest.java [hunk 6]
```diff
@@ -69,7 +69,7 @@ public void noNodesDiscovered() throws Exception {
     @Ignore @Test
     public void nodeDiscovered() throws Exception
     {
-        discoveredNode = new DiscoveredNode("1", com.dell.cpsd.DiscoveredNode.AllocationStatus.DISCOVERED);
+        discoveredNode = new DiscoveredNode("1", com.dell.cpsd.DiscoveredNode.AllocationStatus.DISCOVERED, "testserial", "testproduct","test_vendor");
```
Comment (gavin-scott) [inline dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/delegates/FindDiscoveredNodesTest.java]: space between args

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java (hunk 1 above)
Comment (arunrordell) [reply dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java]: @gavin-scott sorry it is a mistake will update that.

File: dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java [hunk 7]
```diff
@@ -195,13 +195,10 @@ protected void waitForServiceCallback(ServiceCallback serviceCallback, String re
...
-        discoveredNode.setConvergedUuid(uuid);
-        discoveredNode.setAllocationStatus(com.dell.cpsd.DiscoveredNode.AllocationStatus.DISCOVERED);
-
-        com.dell.cpsd.DiscoveredNode addedNode = new com.dell.cpsd.DiscoveredNode();
-        addedNode.setConvergedUuid(UUID.randomUUID().toString());
-        addedNode.setAllocationStatus(com.dell.cpsd.DiscoveredNode.AllocationStatus.ADDED);
+        com.dell.cpsd.DiscoveredNode discoveredNode = new com.dell.cpsd.DiscoveredNode(uuid, "testserial", "testproduct", "test_Dell Inc",
+                com.dell.cpsd.DiscoveredNode.AllocationStatus.DISCOVERED);
+        com.dell.cpsd.DiscoveredNode addedNode = new com.dell.cpsd.DiscoveredNode(UUID.randomUUID().toString(), "testserial",
+                "testproduct", "test_Dell Inc", com.dell.cpsd.DiscoveredNode.AllocationStatus.ADDED);
```
Comment (gavin-scott) [inline dne-paqx/src/test/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeServiceTest.java]: Can you leave these the way they were in lines 198 - 204? That way if the JSON schema for these classes changes this code will still work. I.e. above used the default no-arg constructor which should always be there.

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java [hunk 8]
```diff
@@ -441,25 +443,24 @@ public void executeRequest(String requestId) throws Exception
...
-                DiscoveredNodeInfo discoveredNodeInfo = NodeInventoryParsingUtil.getDiscoveredNodeInfo(nodeInventoryResponse, node.getConvergedUuid());
-                if ( isNodeInventorySaved && discoveredNodeInfo != null) {
-                    discoveredNodeInfo.setNodeStatus(NodeStatus.valueOf(node.getNodeStatus().toString()));
-                    repository.saveDiscoveredNodeInfo(discoveredNodeInfo);
-                    retList.add(discoveredNodeInfo);
-                }
-            } else {
-                LOGGER.info("There is no node inventory for UUID " + node.getConvergedUuid());
-            }
+        if (discoveredNodes == null || discoveredNodes.isEmpty())
```
Comment (gavin-scott) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java]: If it's `null` the loop below on line 457 will throw an NPE. So either go ahead and raise an exception if it's null or do an `else` and don't execute the loop.

File: dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java [hunk 9]
```diff
@@ -441,28 +444,31 @@ public void executeRequest(String requestId) throws Exception
...
-                boolean isNodeInventorySaved = repository.saveNodeInventory(nodeInventory);
-                DiscoveredNodeInfo discoveredNodeInfo = NodeInventoryParsingUtil.getDiscoveredNodeInfo(nodeInventoryResponse, node.getConvergedUuid());
-                if ( isNodeInventorySaved && discoveredNodeInfo != null) {
-                    discoveredNodeInfo.setNodeStatus(NodeStatus.valueOf(node.getNodeStatus().toString()));
-                    repository.saveDiscoveredNodeInfo(discoveredNodeInfo);
-                    retList.add(discoveredNodeInfo);
-                }
-            } else {
-                LOGGER.info("There is no node inventory for UUID " + node.getConvergedUuid());
+        if (discoveredNodes != null && !discoveredNodes.isEmpty())
```
Comment (kushagra-mahajan) [inline dne-paqx/src/main/java/com/dell/cpsd/paqx/dne/service/amqp/AmqpNodeService.java]: No save done, might cause issues.

**Pull request merged**